libs/utils/createBuilder.ts

```
@@ -0,0 +1,37 @@
+import { UserSession } from "@futureverse/auth";
+import { CustomExtrinsicBuilder } from "@futureverse/transact";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+
+import { DEFAULT_GAS_TOKEN } from "../constants";
+
+export async function createBuilder(
+	userSession: UserSession,
+	gasTokenAssetId: number,
+	slippage: string,
+	builder: CustomExtrinsicBuilder,
+	extrinsic: SubmittableExtrinsic<"promise", ISubmittableResult>
+): Promise<CustomExtrinsicBuilder> {
+	builder.reset();
+	const fromEx = builder.fromExtrinsic(extrinsic);
+
+	if (gasTokenAssetId === DEFAULT_GAS_TOKEN.assetId) {
+		try {
+			await fromEx.addFuturePass(userSession.futurepass);
+		} catch (err: any) {
+			console.info(err);
+		}
```

I don't think you want to just catch errors here. You would probably want to let the error propagate and break the app, rather than get into a situation where a user could send an invalid extrinsic